provider/pkg/await/await.go

```diff
@@ -491,9 +499,69 @@ func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 	return currentOutputs, nil
 }
 
+// handleSSAIgnoreFields handles updating the inputs to either drop fields that are present on the cluster and not managed
+// by the current field manager, or to set the value of the field to the last known value applied to the cluster.
+func handleSSAIgnoreFields(c *UpdateConfig, liveOldObj *unstructured.Unstructured) error {
+	managedFields := liveOldObj.GetManagedFields()
+	managedFieldsSet := fieldpath.NewSet()
+
+	for _, f := range managedFields {
+		if f.Manager == c.FieldManager {
+			// We don't need to keep track of fields that we are currently managing.
+			continue
+		}
+		s := fieldpath.NewSet()
+		if err := s.FromJSON(bytes.NewReader(f.FieldsV1.Raw)); err != nil {
+			return fmt.Errorf("unable to parse managed fields from resource %q: %w", c.Inputs.GetName(), err)
+		}
+		managedFieldsSet = managedFieldsSet.Union(s)
+	}
+
+	for _, ignorePath := range c.IgnoreChanges {
+		pathComponents := strings.Split(ignorePath, ".")
+		pe, err := fieldpath.MakePath(makeInterfaceSlice(pathComponents)...)
+		if err != nil {
+			return fmt.Errorf("unable to normalize ignoreField path %q: %w", ignorePath, err)
+		}
+
+		// Drop the field from the inputs if it is present on the cluster and not managed by the current field manager. This ensures
+		// that we don't get any conflict errors, or mistakenly setting the current field manager as a shared manager of that field.
```

Comment:
This comment makes me wonder about cases where pulumi is (or should be) a shared manager.  Would this code cause Pulumi to (inadvertently) relinquish control over a shared field?

Imagine that replicas was `2` and was shared between two managers. The other manager should have to 'force' the conflict to change replicas to `3`. Pulumi shouldn't prematurely relinquish control.

Reply:
Updated the logic to keep track if we're a shared manager or not to prevent this from happening.